packages/boxplot/src/props.ts

```diff
@@ -0,0 +1,82 @@
+import { BoxPlotItem } from './BoxPlotItem'
+import { BoxPlotTooltip, BoxPlotTooltipLabel } from './BoxPlotTooltip'
+import { InheritedColorConfig, OrdinalColorScaleConfig } from '@nivo/colors'
+import { ScaleBandSpec, ScaleSpec } from '@nivo/scales'
+import { ComputedBoxPlotSummary } from './types'
+
+export const defaultProps = {
+    value: 'value',
+    groupBy: 'group',
+    groups: null,
+    subGroupBy: null,
+    subGroups: null,
+
+    quantiles: [0.1, 0.25, 0.5, 0.75, 0.9],
+
+    layout: 'vertical' as const,
+
+    minValue: 'auto' as const,
+    maxValue: 'auto' as const,
+
+    valueScale: { type: 'linear' } as ScaleSpec,
+    indexScale: { type: 'band', round: true } as ScaleBandSpec,
+
+    padding: 0.1,
+    innerPadding: 6,
+
+    axisBottom: {},
+    axisLeft: {},
+    enableGridX: false,
+    enableGridY: true,
+
+    valueFormat: (value: number) => value.toPrecision(4),
+
+    colorBy: 'subGroup' as const,
```

Comment:
Not sure if you're referring to typescript 'dos and donts' or a preference for a default chart look-and-feel. If the former, I'll look into changing this if needed.

If the latter, my justification is as follows. When a chart has a small number of groups and no subgroups, there is no need to color the boxes because all the groups are labeled on the axis. Setting colorBy to 'subGroup' achieves that effect. Also, this default automatically handles more complex datasets that do have subgroups.

One could argue that when there are no subgroups, the chart could give each group a different color. Setting the default to 'groups' would achieve that, but wouldn't be appropriate for more complex datasets with subgroups. So to handle both cases, we would need to have 'colorBy' as null by default and set its value at run time.

So, would you prefer the default as 'subGroup' or 'null'? I'm OK either way.

Reply:
>. When a chart has a small number of groups and no subgroups, there is no need to color the boxes because all the groups are labeled on the axis. Setting colorBy to 'subGroup' achieves that effect. Also, this default automatically handles more complex datasets that do have subgroups.

Makes sense, thx for the explanation.